Automattic/wp-calypso
# Add/support page block

#### Proposed Changes

* Adds Gutenberg block to embed support page according to the design outlined in 78-gh/Automattic/lighthouse-forums

Things that will be implemented in a separate PR:
- Display the number of people that find the page useful

#### Testing Instructions

* Run `yarn dev --sync` in `apps/happy-blocks` to sync code to your sandbox
* On your sandbox, edit `wp-content/mu-plugins/wpcom-bbpress.php` and add these lines
```
		if ( file_exists( WP_CONTENT_DIR . '/a8c-plugins/happy-blocks/index.php' ) ) {
			require_once WP_CONTENT_DIR . '/a8c-plugins/happy-blocks/index.php';
		}
```
to the function `wpcom_bbpress_load_plugins` to load `happy-blocks` plugin.
* On your sandbox, edit `wp-content/themes/a8c/supportforums/editor/gutenberg-wpcom.php` and add to the function `supportforums_gutenberg_setup_frontend` before `return $settings;` line:
```
$settings['iso']['blocks']['allowBlocks'][] = 'happy-blocks/support-page';
```
* Create a new topic, in the editor press `/` and search for "WordPress Guide" block.
* Insert support page URL, for example, https://wordpress.com/support/domains/change-a-domain/, and verify its content has been embedded in the editor.
* Save the topic, and verify embedded block was rendered in the topic view.
* Verify embed is inserted by pasting the support page URL directly into the editor.

![image](https://user-images.githubusercontent.com/133608/201077744-3b2629e7-9153-464a-bc0b-41d998af5ea1.png)

Note, that after merging PR D90725-code it should be possible to install it using Calypso 'install-plugin.sh`

#### Pre-merge Checklist
